extension/src/main/webapp/WEB-INF/conf/gamification/storage/rule/rules-storage-configuration.xml

```diff
@@ -29,8 +29,15 @@ Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
     <component>
         <type>org.exoplatform.addons.gamification.service.configuration.RuleService</type>
     </component>
-
-
+    <!-- Challenge storage-->
+    <component>
+        <type>org.exoplatform.addons.gamification.storage.ChallengeStorage</type>
+    </component>
+    <!-- Challenge Service Layer-->
+    <component>
+        <key>org.exoplatform.addons.gamification.service.ChallengeService</key>
+        <type>org.exoplatform.addons.gamification.service.configuration.ChallengeServiceImpl</type>
+    </component>
```

Comment:
delete this please is already added in completion-configuration.xml!!

Reply:
just reorganized each service in his related XML file!